manifests/overrides/README.md

```diff
@@ -0,0 +1,5 @@
+# Overridable Manifests
+
+These are manifest files that are intended to be used as-is or to be overridden.
+
+> **Note:** There should be note root `kustomization.yaml` as each folder should be treated as a specific target for inclusion or overrides.
```

Comment:
I get the new structure, and I'm not against it given our particular set of deployments, but I find the override folder a little bit misleading, kustomize is set on the premise of [bases and overlays](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays). You tend to find that kind of structure so this might confuse newcomers that are used to the common kustomize flow.

Reply:
I'll respond here since these threads seem related and GitHub sucks at root-level conversations 🙂 

> And my other comment is that I find that both `common` and `overrides` folder seem to be the `base` files from manifest and `odh`, `rhoai/addon`, `rhoai/onprem` as overlays. I get the difference, but why don't label the first two under a `base` folder and the other three under a `overlay`?
> 
> It's just semantics but might stick to what kustomize suggests organizing the files.

([from here](https://github.com/opendatahub-io/odh-dashboard/pull/2906#issuecomment-2186648170))

@lucferbux The way we used overlays were highly confusing to me, so I did away with it in the first pass. If we find having overlays to be specifically beneficial, we can craft those as needed and maybe create a new overlays folder; I didn't see anything useful in the last one and deleted them in the restructure effort.

We don't want structure out manifests to be "layers on top of each other" -- since there are 3 semi-unique entry points. It's not a matter of "a is the base" "b is additions on that"... the way we had it was "overlays were a mixed bag of RHOAI manifests and optional overlays"; I'm not sure this was the intended design of kustomize's "bases and overlays".

I'll think on if there is a better way to shape the use of overlays -- but honestly, it feels like we have a use-case that doesn't work for that design and for greater management and simplicity, we do away with it. I appreciate the input, but our old usage of `overlays` felt confusing and left a lot of questions about "what is what" and "is this for x deployment". Overlays always struck me as "additions to an existing setup" and not "ways to configure multiple different core deployment types". I feel overlays need to speak to "optional opt-ins" and not core fundamental stacking points.

Happy to discuss this further if you feel this is a confusing setup for `kustomize` experts. Maybe there is a middle ground here, but I don't currently see a need for overlays.

Reply:
Also, I forgot to address the other point -- `overrides` is perhaps not the best name -- I could call them `overlays` but I feel that's still confusing use of `overlays`. I'm open to discussing alternatives here.

Reply:
The text on the kustomize website definitely feels like maybe we call `overrides` something like "custom-bases" or "core-bases" -- and that sorta fits the design overlap... but it'll be "different" than kustomizes suggestion.

Reply:
Okay, I have updated it somewhat based on your feedback... @lucferbux I still don't believe we need to be so basic with our definitions... `bases` & `overlays` is a naive structure that makes it difficult to manage 3 deployments in a way that is both readable and functional. If that causes friction with kustomize experts, then it is a price to pay for maintainability imo.

Reply:
Ok, that's fair tbh, we can keep that structure, it's fine to me having this layout.